src/mmw/apps/geoprocessing_api/validation.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+import json
+
+from numbers import Number
+
+from django.conf import settings
+from django.contrib.gis.geos import GEOSGeometry
+from rest_framework.exceptions import ValidationError
+
+
+def create_invalid_rwd_location_format_error_msg(loc):
+    return ('Invalid required `location` parameter value `{}`. Must be a '
+            '`[lat, lng] where `lat` and `lng` are numeric.'.format(loc))
+
+
+def validate_location_format(loc):
+    if loc is None or type(loc) is not list or len(loc) is not 2:
+        return False
+    else:
+        [lat, lng] = loc
+        return (isinstance(lat, Number) and isinstance(lng, Number) and
+                type(lat) is not bool and type(lng) is not bool)
+
+
+def create_invalid_rwd_data_source_error_msg(source):
+    return ('Invalid optional `dataSource` parameter value `{}`. Must be '
+            '`drb` or `nhd`.'.format(source))
+
+
+def validate_rwd_data_source(source):
+    return source in ['drb', 'nhd']
+
+
+def create_invalid_rwd_snapping_error_msg(snapping):
+    return ('Invalid optional `snappingOn` parameter value `{}`. Must be '
+            '`true` or `false`.'.format(snapping))
+
+
+def create_invalid_rwd_simplify_param_type_error_msg(simplify):
+    return ('Invalid optional `simplify` parameter value: `{}`. Must be a '
+            'number.'.format(simplify))
+
+
+def validate_rwd_simplify_param_type(simplify):
```

Comment:
Consider creating a single function `validate_rwd_request_payload` that checks all parts of the body for errors, and raises a validation error with the list of things that are wrong.